### PR TITLE
Add multiplayer support & Misc Changes

### PR DESCRIPTION
**Pull request by TheJaysH**
* Added support for Multiplayer
* Added Block Blacklist
    *  this will allow users to define a list of blocks to not mine (that appear in tags)
* Fixed Bugs

### CODE REVIEW

**Comment by TheJaysH** [comment]
@altaiiiir Please see changes above. havent tested in MP yet, but should work fine. refactored the mod structure a bit to make it easier if/when more is added.  and added a feature request for a blacklist

**Comment by TheJaysH** [comment]
Dont merge yet, have a couple more changes Managed to get an outline rendering.

image

File: gradle.properties [hunk 1]
```diff
@@ -48,7 +48,7 @@ mod_name=Viner
 # The license of the mod. Review your options at https://choosealicense.com/. All Rights Reserved is the default.
 mod_license=MIT License
 # The mod version. See https://semver.org/
-mod_version=1.4.0
+mod_version=1.5.0
```
Comment (altaiiiir) [inline gradle.properties:51]: With the addition of the Multiplayer fixes (thanks) I'm planning to make the last PR. this one, as well as some more tweaks on my end and releasing it on Modrinth as 2.0.0. 

If you have a modrinth profile let me know so I can credit you in the release notes!

**Comment by altaiiiir** [comment]
Really appreciate the help @TheJaysH. The plan is to merge this PR (without the outline rendering) and release it as 2.0.0. 

Unsure that I like the outline to be completely honest, I like the vanilla feeling of this mod. Again, thanks a lot for your help on this. Probably looking to merge later today.

File: src/main/java/com/ael/viner/config/Config.java [hunk 2]
```diff
@@ -44,18 +48,27 @@ public class Config {
...
+                        obj -> obj instanceof String && ((String) obj).matches("^[a-z_]+:[a-z_]+$"));
+
         BUILDER.pop();
 
         SPEC = BUILDER.build();
     }
 
     @SubscribeEvent
     public static void onLoadComplete(FMLLoadCompleteEvent event) {
-//        upgradeConfig();
```
Comment (altaiiiir) [inline src/main/java/com/ael/viner/config/Config.java:58]: Was curious what this was for, I see now. 

Pretty good idea for user experience!

File: src/main/java/com/ael/viner/util/MiningUtils.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+            // Getting block state of connected block
+            BlockState blockState = level.getBlockState(blockPos);
+
+            // Checking for Silk Touch enchantment
+            boolean hasSilkTouch = EnchantmentHelper.getTagEnchantmentLevel(Enchantments.SILK_TOUCH, tool) > 0;
+
+            if (hasSilkTouch) {
+                Block.popResource(level, firstBlockPos, new ItemStack(blockState.getBlock()));
+            } else {
+                // FIXME: We need to implement logic for Fortune
```
Comment (altaiiiir) [inline src/main/java/com/ael/viner/util/MiningUtils.java:50]: Implemented a temp workaround for this yesterday by trying to replicate Vanilla fortune logic as close as possible

**Review by altaiiiir — APPROVED** [review]
Thanks for this PR! Please create a separate ticket for the block outlines if you really think people would enjoy it.

Please make it toggleable and false by default if anything! Also not sure if I mentioned this already but please provide your Modrinth account so I can credit you for all your great help! much appreciated

**Comment by TheJaysH** [comment]
I'm in no rush to finish the block outline. Was just something I felt it needed. It's been fun to work on. 

It's not always clear what blocks will be mined. The outline just makes it clear to the user.

**Pull request merged**